pykafka/balancedconsumer.py

```diff
@@ -535,7 +566,24 @@ def _rebalance(self):
 
                     # Only re-create internal consumer if something changed.
                     if new_partitions != self._partitions:
-                        self._setup_internal_consumer(list(new_partitions))
+                        cns = self._get_internal_consumer(list(new_partitions))
+                        if self._consumer is not None and \
+                                self._partitions_revoked_callback is not None:
+                            self._partitions_revoked_callback(
+                                self, _get_held_offsets_for_partitions(
+                                    current_zk_parts - new_partitions, self._consumer))
+                        if self._partitions_assigned_callback is not None:
+                            new_offsets = self._partitions_assigned_callback(
+                                self,
+                                _get_held_offsets_for_partitions(new_partitions, cns))
```

Comment:
Apologies for the sort of stream-of-consciousness comment flood. I'm now thinking, the JVM consumer defines two callbacks because they call them in two different states, but now that we call these back-to-back.... so wouldn't we provide the same utility if we had.... one callback, with signature `rebal_callback(self, all_old_partitions_offsets, all_current_partitions_offsets ) => {returning optionally a new set of offsets}`?

That way users would get access to all the offsets, from before and after, and if they wanted to know which ones had been dropped / newly assigned, they could easily diff the two sets.

Sorry for all the hemming and hawing - I'm actually quite excited about how powerful this interface is becoming!!